### PR TITLE
TEXT-40 : Escape HTML characters once

### PR DESCRIPTION
**Pull request by SampannaKahu**
Hi, this is my first time contributing to any open source project.
The feature goes as follows:

If already escaped HTML characters are in the input text, StringEscapeUtils.escapeHtml4() escapes them again.

For example:
If the input is:
100 kg \&lt; 1000kg
Then the output of escapeHtml4() becomes:
100kg \&amp;lt; 1000kg

At my workplace, we felt the need for a method in StringEscapeUtils which does not escape already escaped characters.

### CODE REVIEW

**Comment by SampannaKahu** [comment]
Facing issues with the build. Working on it.

**Comment by SampannaKahu** [comment]
Build is succeeding on my local using the command run by travis-ci (mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V).
Closing and re-opening the pull request to re-trigger the build.

**Comment by chtompki** [comment]
I'll work on pulling this in. I may make some changes before pulling it in, but if I think that anything of any substance needs changing, I'll let you know so you can double check what I've done.

**Comment by SampannaKahu** [comment]
Sure, thanks.

File: src/main/java/org/apache/commons/text/StringEscapeUtils.java [hunk 1]
```diff
@@ -197,6 +197,27 @@
...
+     * object allows the HTML escaping functionality to be used
+     * as the foundation for a custom translator.
+     *
+     * Note that, multiple lookup tables should be passed to this translator
+     * instead of passing multiple instances of this translator to the
+     * AggregateTranslator. Because, a SingleLookupTranslator only checks the values of the
+     * lookup table passed to that instance while deciding whether a value is
+     * already translated or not.
+     *
+     * @since 3.0
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/StringEscapeUtils.java]: Going to remove the `@since 3.0` from the javadocs.

**Comment by chtompki** [comment]
I'm also going to add you to the `pom.xml` under contributors and add your change to the `changes.xml` as those are the way we keep track of what goes in a release.

**Comment by chtompki** [comment]
You mind if I add the following unit test?

```java
@Test
    public void testEscapeHtml3Once() {
        for (final String[] element : HTML_ESCAPES) {
            final String message = element[0];
            final String expected = element[1];
            final String original = element[2];
            assertEquals(message, expected, StringEscapeUtils.escapeHtml4Once(original));
            assertEquals(message, expected, StringEscapeUtils.escapeHtml4Once(expected));
            final StringWriter sw = new StringWriter();
            try {
                StringEscapeUtils.ESCAPE_HTML3_ONCE.translate(original, sw);
            } catch (final IOException e) {
            }
            final String actual = original == null ? null : sw.toString();
            assertEquals(message, expected, actual);
            final StringWriter sw2 = new StringWriter();
            try {
                StringEscapeUtils.ESCAPE_HTML3_ONCE.translate(expected, sw2);
            } catch (final IOException e) {
            }
            final String actual2 = original == null ? null : sw2.toString();
            assertEquals(message, expected, actual2);
        }
    }
```

**Comment by SampannaKahu** [comment]
Please go ahead, if it is passing

**Comment by chtompki** [comment]
Cool. Will do.

**Comment by chtompki** [comment]
If you merge master into your branch and push that up to your branch `escape_html_once`, the PR should show as merged.

**Comment by chtompki** [comment]
On behalf of all of us on the Apache Commons Project, many thanks for your contribution.

From me personally, we need more folks out there like you that are willing to help. So many thanks and cheers.

**Comment by chtompki** [comment]
there we go.

**Comment by SampannaKahu** [comment]
Thanks a lot Rob!
Glad I could help 😄

**Pull request merged**